cross-chain/arbitrum/hardhat.config.ts

```diff
@@ -0,0 +1,120 @@
+import type { HardhatUserConfig } from "hardhat/config"
+
+import "@keep-network/hardhat-helpers"
+import "@keep-network/hardhat-local-networks-config"
+import "@nomiclabs/hardhat-waffle"
+import "@nomiclabs/hardhat-etherscan"
+import "hardhat-gas-reporter"
+import "hardhat-contract-sizer"
+import "hardhat-deploy"
+import "@typechain/hardhat"
+import "hardhat-dependency-compiler"
+
+const config: HardhatUserConfig = {
+  solidity: {
+    compilers: [
+      {
+        version: "0.8.17",
+        settings: {
+          optimizer: {
+            enabled: true,
+            runs: 1000,
+          },
+        },
+      },
+    ],
+  },
+
+  paths: {
+    artifacts: "./build",
+  },
+
+  networks: {
+    hardhat: {
+      deploy: ["deploy_l2"],
+    },
+    goerliEthereum: {
+      url: process.env.CHAIN_L1_API_URL || "",
+      chainId: 5,
+      deploy: ["deploy_l1"],
+      accounts: process.env.ACCOUNTS_L1_PRIVATE_KEYS
+        ? process.env.ACCOUNTS_L1_PRIVATE_KEYS.split(",")
+        : undefined,
+      tags: ["etherscan"],
+    },
+    mainnetEthereum: {
+      url: process.env.CHAIN_L1_API_URL || "",
+      chainId: 1,
+      deploy: ["deploy_l1"],
+      accounts: process.env.ACCOUNTS_L1_PRIVATE_KEYS
+        ? process.env.ACCOUNTS_L1_PRIVATE_KEYS.split(",")
+        : undefined,
+      tags: ["etherscan"],
+    },
+    goerliArbitrum: {
+      url: process.env.CHAIN_L2_API_URL || "",
+      chainId: 421613,
+      deploy: ["deploy_l2"],
+      accounts: process.env.ACCOUNTS_L2_PRIVATE_KEYS
+        ? process.env.ACCOUNTS_L2_PRIVATE_KEYS.split(",")
+        : undefined,
+      tags: ["arbiscan"],
+      companionNetworks: {
+        l1: "goerliEthereum",
+      },
```

Comment:
I am reading through https://github.com/wighawag/hardhat-deploy#companionnetworks and I am still not sure what is this option doing. Can you shed some light, please? Also, why don't we set something similar for mainnet configs?

Reply:
This configuration was added for the future [extended implementation explained in RFC-8](https://github.com/keep-network/tbtc-v2/blob/main/docs/rfc/rfc-8.adoc#372-extended-implementation). Companion network allows us to deploy contracts on multiple networks L1 and L2 by running just one command: `yarn deploy (options)`. In our case, it will look at the `deploy_l1` and `deploy_l2` dirs and execute the deployment scripts that are in these dirs. Technically we can use just the `*Arbitrum` network without referencing L1, but I thought that we can prepare the configuration especially since it's written in the RFC-8.
Companion network is present for both Arbitrum goerli and mainnet.